theano/gof/destroyhandler.py

```diff
@@ -945,6 +945,20 @@ def orderings(self, fgraph):
                 raise InconsistencyError("Attempting to destroy indestructible variables: %s" %
                         illegal_destroy)
 
+            # We want to compute a list of nodes that may never be computed,
+            # so that we do not force their computations (they may be uncomputeable).
+            # For that, we start by a list of lazy nodes (ifelse), and use io_toposort
+            # to find out which variables will always be computed, even if the lazy
+            # ones are not.
+            lazy_outputs = [var for var in fgraph.variables
+                            if var.owner and isinstance(var.owner, theano.ifelse.IfElse)]
+            always_computed = graph.io_toposort(fgraph.outputs, lazy_outputs)
```

Comment:
I just don't understand this line. If there is only one ifelse, I think it will return all the node needed for the 3 inputs of the ifelse.

I think this PR "fix" the problem as the line 956 check if a variable is in the list of apply nodes. So not_always_computed contain all variable. So it mostly disable all node with destroy_map. So it disable all inplace.